docs/architecture/architecture.md

```diff
@@ -0,0 +1,129 @@
+# Architecture
+
+This document describes the architecture of `ibc-rs`. If you're looking for a high-level overview of the code base, you've come to the right place!
+
+## Terms 
+
+Some important terms and acronyms that are commonly used include:
+
+ * **IBC**: Refers to the **I**nter**B**lockchain **C**ommunication protocol, a distributed protocol that allows different sovereign blockchains to communicate with one another. The protocol has both on-chain and off-chain components.
+ * **ICS**: Refers to **I**nter**C**hain **S**tandards, which are stadardization documents that capture the specifications of the IBC protocol across multiple documents. For example, ICS02 captures the client abstraction of the IBC protocol.
+ * **IBC module**: Refers to a piece of on-chain logic on an IBC-enabled chain.
+ * **Relayer**: Refers to an off-chain process that is responsible for relaying packets between chains.
+ * **Hermes**: Refers to the `ibc-rs` crate's particular relayer implementation. 
+
+## Bird's Eye View
+
+![][layout-image]
```

Comment:
Note that in this image the `tendermint-rs` box refers to a consensus engine. It may be inaccurate to depict it that way. Briefly, the "consensus" box belongs somewhere outside of this diagram.

For more context:
1. `tendermint-rs` includes no implementation of consensus, it comprises mostly data types and serialization tools to interface with a consensus engine;
    - consensus is implemented in [`tendermint/tendermint`](https://github.com/tendermint/tendermint), and `tendermint-rs` merely provides a way to interface via (g)RPC with a process running tendermint/tendermint.
2. `ibc-relayer` fetches block headers from chains and verifies them;
    - suppposing a client-server model, `tendermint-rs` is the client and acts as a library, whereas `tendermint/tendermint` is the server; in this way, `ibc-rs` is using the client library to fetch (and verify) headers from the server.

Reply:
Could we perhaps just re-label the `tendermint-rs` box to `Tendermint`? That way the whole notion of the consensus engine is abstracted away, which I think is fine since that's not the focus here in the diagram.

Reply:
Hmm, we could relabel it, but then I would suggest the box should look differently. Currently, blue boxes with underline text means _rust crate_. If we re-label, that will no longer be the case.

In fact, instead of putting into the box `Tendermint`, I would suggest the label should be `IBC-enabled chain (e.g., based on Cosmos SDK)`

Reply:
Saw the update, looks good!

Reply:
The image has transparent background, some elements are not visible with a Dark theme on GitHub or any other viewer that might use a dark background.